src/Servers/IIS/IIS/test/Common.FunctionalTests/ServerAbortTests.cs

```diff
@@ -4,51 +4,52 @@
 using System.Net;
 using System.Net.Http;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IIS.FunctionalTests.Utilities;
 using Microsoft.AspNetCore.Server.IntegrationTesting;
 using Microsoft.AspNetCore.Testing.xunit;
 using Xunit;
 
 namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
 {
-    public abstract class ServerAbortTests: FixtureLoggedTest
+    [Collection(PublishedSitesCollection.Name)]
+    public class ServerAbortOutOfProcessTests : IISFunctionalTestBase
     {
-        private readonly IISTestSiteFixture _fixture;
-
-        [Collection(IISTestSiteCollection.Name)]
-        public class InProc: ServerAbortTests
+        public ServerAbortOutOfProcessTests(PublishedSitesFixture fixture) : base(fixture)
         {
-            public InProc(IISTestSiteFixture fixture) : base(fixture) { }
         }
 
-        [Collection(OutOfProcessTestSiteCollection.Name)]
-        public class OutOfProcess: ServerAbortTests
+        [ConditionalFact]
+        public async Task ClosesConnectionOnServerAbortOutOfProcess()
         {
-            public OutOfProcess(OutOfProcessTestSiteFixture fixture) : base(fixture) { }
-        }
+            try
+            {
+                var deploymentParameters = Fixture.GetBaseDeploymentParameters(HostingModel.OutOfProcess);
 
-        protected ServerAbortTests(IISTestSiteFixture fixture) : base(fixture)
-        {
-            _fixture = fixture;
+                var deploymentResult = await DeployAsync(deploymentParameters);
+
+                var response = await deploymentResult.HttpClient.GetAsync("/Abort").DefaultTimeout();
+
+                Assert.Equal(HttpStatusCode.BadGateway, response.StatusCode);
+                // 0x80072f78 ERROR_HTTP_INVALID_SERVER_RESPONSE The server returned an invalid or unrecognized response
+                Assert.Contains("0x80072f78", await response.Content.ReadAsStringAsync());
+            }
+            catch (HttpRequestException)
```

Comment:
Is this catch needed?

Reply:
Yeah. OutOfProc can either fail from an exception being throw or a bad gateway.